File: src/main/java/no/nav/personopplysninger/features/endreopplysninger/domain/adresse/Stedsadresse.java
package no.nav.personopplysninger.features.endreopplysninger.domain.adresse;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Stedsadresse {
    @JsonProperty("@type")
    private String subtype = "NORSK_STEDSADRESSE";

    private String kilde = "BRUKER SELV";
    private String eiendomsnavn;
    private String gyldigTom;
    private String kommunenummer;
    private String postnummer;
    private String tilleggslinje;
    private String tilleggslinjeType;

    @JsonCreator
    public static Stedsadresse create(String json) throws JsonParseException, JsonMappingException, IOException {
        return (new ObjectMapper()).readValue(json, Stedsadresse.class);
    }

    public String getKilde() {
        return kilde;
    }

    public String getEiendomsnavn() {
        return eiendomsnavn;
    }

    public String getGyldigTom() {
        return gyldigTom;
    }

    public String getKommunenummer() {
        return kommunenummer;
    }

    public String getPostnummer() {
        return postnummer;
    }

    public String getTilleggslinje() {
        return tilleggslinje;
    }

    public String getTilleggslinjeType() {
        return tilleggslinjeType;
    }
}

File: src/main/kotlin/no/nav/personopplysninger/features/endreopplysninger/domain/adresse/Stedsadresse.kt
package no.nav.personopplysninger.features.endreopplysninger.domain.adresse

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.core.JsonParseException
import com.fasterxml.jackson.databind.JsonMappingException
import com.fasterxml.jackson.databind.ObjectMapper

import java.io.IOException

@JsonIgnoreProperties(ignoreUnknown = true)
class Stedsadresse {
    @JsonProperty("@type")
    private val subtype = "NORSK_STEDSADRESSE"

    val kilde = "BRUKER SELV"
    val eiendomsnavn: String? = null
    val gyldigTom: String? = null
    val kommunenummer: String? = null
    val postnummer: String? = null
    val tilleggslinje: String? = null
    val tilleggslinjeType: String? = null

    companion object {

        @JsonCreator
        @JvmStatic
        @Throws(JsonParseException::class, JsonMappingException::class, IOException::class)
        fun create(json: String): Stedsadresse {
            return ObjectMapper().readValue(json, Stedsadresse::class.java)
        }
    }
}
